src/Cassandra/Connections/Connection.cs

```diff
@@ -555,14 +592,21 @@ internal bool ReadParse(byte[] buffer, int length)
                     break;
                 }
                 stream = stream ?? Configuration.BufferPool.GetStream(Connection.StreamReadTag);
-                var state = header.Opcode != EventResponse.OpCode
-                    ? RemoveFromPending(header.StreamId)
-                    : new OperationState(EventHandler);
+                // todo(sivukhin, 14.04.2019): Is it crucial to remove state from pending operations strictly before writing remaining body to the stream?
```

Comment:
Ideally the remove operation needs to happen as fast as possible because another thread might be trying to cancel (e.g. timeout) this request when we already have the response.